templates/CRM/Profile/Form/Dynamic.tpl

```diff
@@ -148,7 +148,7 @@
               {elseif $n|substr:0:5 eq 'phone'}
                 {assign var="phone_ext_field" value=$n|replace:'phone':'phone_ext'}
                 {$form.$n.html}
-                {if $form.$phone_ext_field.html}
+                {if !empty($form.$phone_ext_field.html)}
```

Comment:
For these ones, when it's difficult to ensure it's assigned on the php side, we've been using array_key_exists, like
`{if array_key_exists($phone_ext_field, $form)}`

Reply:
(Note normally would also check that it has a value, i.e. `{if array_key_exists($phone_ext_field, $form) && $form.$phone_ext_field}`, but it's not necessary for quickform fields since they are either there or not, and if they're there, then they'll also have html.)